conjure-core/src/main/java/com/palantir/conjure/parser/services/EndpointError.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.parser.services;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.palantir.conjure.defs.ConjureImmutablesStyle;
+import com.palantir.conjure.parser.services.EndpointError.EndpointErrorDeserializer;
+import com.palantir.conjure.parser.types.ConjureType;
+import com.palantir.parsec.ParseException;
+import java.io.IOException;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@JsonDeserialize(using = EndpointErrorDeserializer.class)
+@Value.Immutable
+@ConjureImmutablesStyle
+public interface EndpointError {
+    @JsonProperty("error-name")
```

Comment:
if we make the `errorName`->`error` change suggested above, we should also call this `error`

Reply:
We should update the json-schema to support the new fields as well: https://github.com/palantir/conjure/blob/master/conjure.schema.json

Reply:
I think there's a test that we should be able to make fail until that's updated

Reply:
There was a failing test (`sort_types_services_and_errors`), but I don't think it references the conjure.schema.json file. I was able to recreate the `normalized.conjure.json` file and make the test pass, even though I didn't yet update the `conjure.schema.json` file.

I'll checkout how to validate a conjure definition against the JSON schema in a test.

Reply:
Just needed to add an endpoint error to [this file](https://github.com/palantir/conjure/blob/master/conjure/src/test/resources/test-service.yml) to make it fail validation.